lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -0,0 +1,259 @@
+                           priority=priority, goto_table=self.next_table)
+       
+        # Add ARP flow for LOCAL port
+        if ue_ip_adr is not None:
+            match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,in_port=self._uplink_port,
+                               arp_tpa=ue_ip_adr)
+            if sid:
+                actions = [parser.OFPActionSetField(metadata=sid)]
+            else:
+                actions = []
+
+            flows.add_flow(self._datapath, self.tbl_num, match, actions=actions,
+                           priority=priority, goto_table=self.next_table)
+
+        # Add ARP flow for mtr port
+        if ue_ip_adr is not None:
+            match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                               in_port=self.config.mtr_port,
+                               arp_tpa=ue_ip_adr)
+            if sid:
+                actions = [parser.OFPActionSetField(metadata=sid)]
+            else:
+                actions = []
+
+            flows.add_flow(self._datapath, self.tbl_num, match, actions=actions,
+                           priority=priority, goto_table=self.next_table)
+      
+
+    def _delete_gtp_tunnel_flows(self, i_teid:int, ue_ip_adr:str):
+        
+        # Delete flow for gtp port
+        if i_teid:
+            match = MagmaMatch(tunnel_id=i_teid, in_port=self.config.gtp_port)
+            flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete flow for LOCAL port
+        if ue_ip_adr is not None:
+            match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                               in_port=self._uplink_port, ipv4_dst=ue_ip_adr)
+            flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete flow for mtr port
+        if ue_ip_adr is not None:
+            match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                               in_port=self.config.mtr_port,ipv4_dst=ue_ip_adr)
+            flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete ARP flow for LOCAL port
+        if ue_ip_adr is not None:
+            match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                               in_port=self._uplink_port, arp_tpa=ue_ip_adr)
+            flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete ARP flow for mtr port
+        if ue_ip_adr is not None:
+            match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                               in_port=self.config.mtr_port, arp_tpa=ue_ip_adr)
+            flows.delete_flow(self._datapath, self.tbl_num, match)
+
+
+    def _add_discard_data_gtp_tunnel_flows(self, precedence:int, i_teid:int,
+                                            ue_ip_adr:str):
+        priority = self.convert_precedence_to_priority(precedence)
+        # discard uplink Tunnel
+        if i_teid:
+            match = MagmaMatch(tunnel_id=i_teid, in_port=self.config.gtp_port)
+            cookie = self._cookie
```

Comment:
As OAI code, cookie is used for forward_downlink_tunnel_flow and discard_tunnel_flow. So that we are added here.
But i am not sure about the use-cases.